xmltodict.py

```diff
@@ -70,13 +70,16 @@ def __init__(self,
         self.force_list = force_list
 
     def _build_name(self, full_name):
-        if not self.namespaces:
+        if self.namespaces is None:
             return full_name
         i = full_name.rfind(self.namespace_separator)
         if i == -1:
             return full_name
         namespace, name = full_name[:i], full_name[i+1:]
-        short_namespace = self.namespaces.get(namespace, namespace)
+        try:
+            short_namespace = self.namespaces[namespace]
+        except KeyError:
+            short_namespace = namespace
```

Comment:
From the [documentation](https://docs.python.org/3.7/library/collections.html#collections.defaultdict) on `defaultdict`:

> Note that `__missing__()` is not called for any operations besides `__getitem__()`. This means that `get()` will, like normal dictionaries, return `None` as a default rather than using `default_factory`.

While a bit more verbose, this change uses `__getitem__()` instead of `get()`, so the `defaultdict` behaves as expected.